app/assets/javascripts/pretty-text/engines/discourse-markdown/hashtag-autocomplete.js

```diff
@@ -78,10 +78,24 @@ function addHashtag(buffer, matches, state) {
 // Instead, the UI will dynamically replace these where hashtags
 // are rendered, like within posts, using decorateCooked* APIs.
 function addIconPlaceholder(buffer, state) {
-  const token = new state.Token("span_open", "span", 1);
+  let token = new state.Token("span_open", "span", 1);
   token.block = false;
   token.attrs = [["class", "hashtag-icon-placeholder"]];
   buffer.push(token);
+
+  token = new state.Token("svg_open", "svg", 1);
+  token.block = false;
+  token.attrs = [["class", `fa d-icon d-icon-square-full svg-icon svg-node`]];
```

Comment:
This is just so we can show something before the `decorateCooked` calls take effect on the client; for posts this doesn't seem to be an issue as we appear to hide the content of the posts until they are cooked...